src/odemis/dataio/hdf5.py

```diff
@@ -1200,132 +1030,72 @@ def _add_image_metadata(group, image, mds):
-        else:
-            posqwp.append(0)
-            st_posqwp.append(ST_INVALID)
-        if model.MD_POL_POS_LINPOL in md:
-            poslinpol.append(md[model.MD_POL_POS_LINPOL])
-            st_poslinpol.append(ST_REPORTED)
-        else:
-            poslinpol.append(0)
-            st_poslinpol.append(ST_INVALID)
-
+        get_status(st_pol, pol, md, model.MD_POL_MODE)
+        get_status(st_posqwp, posqwp, md, model.MD_POL_POS_QWP, default=False)
+        get_status(st_poslinpol, poslinpol, md, model.MD_POL_POS_LINPOL, default=False)
         # streak camera
-        if model.MD_STREAK_TIMERANGE in md:
-            timeRange.append(md[model.MD_STREAK_TIMERANGE])
-            st_timeRange.append(ST_REPORTED)
-        else:
-            timeRange.append("")
-            st_timeRange.append(ST_INVALID)
-        if model.MD_STREAK_MCPGAIN in md:
-            MCPGain.append(md[model.MD_STREAK_MCPGAIN])
-            st_MCPGain.append(ST_REPORTED)
-        else:
-            MCPGain.append("")
-            st_MCPGain.append(ST_INVALID)
-        if model.MD_STREAK_MODE in md:
-            streakMode.append(md[model.MD_STREAK_MODE])
-            st_streakMode.append(ST_REPORTED)
-        else:
-            streakMode.append("")
-            st_streakMode.append(ST_INVALID)
-        if model.MD_TRIGGER_DELAY in md:
-            triggerDelay.append(md[model.MD_TRIGGER_DELAY])
-            st_triggerDelay.append(ST_REPORTED)
-        else:
-            triggerDelay.append("")
-            st_triggerDelay.append(ST_INVALID)
-        if model.MD_TRIGGER_RATE in md:
-            triggerRate.append(md[model.MD_TRIGGER_RATE])
-            st_triggerRate.append(ST_REPORTED)
+        get_status(st_timeRange, timeRange, md, model.MD_STREAK_TIMERANGE)
+        get_status(st_MCPGain, MCPGain, md, model.MD_STREAK_MCPGAIN)
+        get_status(st_streakMode, streakMode, md, model.MD_STREAK_MODE)
+        get_status(st_triggerDelay, triggerDelay, md, model.MD_TRIGGER_DELAY)
+        get_status(st_triggerRate, triggerRate, md, model.MD_TRIGGER_RATE)
+
+    # integration time
+    set_status(gp, "IntegrationTime", st_its, its)
+    # angle resolved
+    set_status(gp, "PolePosition", st_pp, pp)
+    set_status(gp, "XMax", st_xm, xm)
+    set_status(gp, "HoleDiameter", st_hd, hd)
+    set_status(gp, "FocusDistance", st_fd, fd)
+    set_status(gp, "ParabolaF", st_pf, pf)
+    # polarization analyzer
+    set_status(gp, "Polarization", st_pol, pol)
+    set_status(gp, "QuarterWavePlate", st_posqwp, posqwp)
+    set_status(gp, "LinearPolarizer", st_poslinpol, poslinpol)
+    # streak camera
+    set_status(gp, "TimeRange", st_timeRange, timeRange)
+    set_status(gp, "MCPGain", st_MCPGain, MCPGain)
+    set_status(gp, "StreakMode", st_streakMode, streakMode)
+    set_status(gp, "TriggerDelay", st_triggerDelay, triggerDelay)
+    set_status(gp, "TriggerRate", st_triggerRate, triggerRate)
+
+
+def get_status(status_list, value_list, md, MD_key, default=True):
```

Comment:
`get_status()` is quite confusing name. Maybe `append_value` or `append_metadata`, or`prepare_...`. At least, it shouldn't start with `get_`, as it doesn't return any value, and avoid `status`, as it's too close from "state".

Also, the current `default` behaviour is confusing. `True` means that it'll set an empty string, and False means it'll set 0?! It'd suggest to just get the default value, so it would be `default=""`, and some callers would use `default=0` (or other values).

BTW, don't forget the docstring ;-)